### ISSUE TITLE
Support for Recovery Services Vault notification preferences

### ISSUE THREAD

**Issue by elliot-resdiary** [terraform-providers/terraform-provider-azurerm#5423]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It's great that we have the resources to create Recovery Services Vaults, protection policies, and VM protection, however, arguably just as important as the backups themselves are the email notification preferences which let us know when backups aren't running or have been deleted.

### New or Affected Resource(s)

* azurerm_recovery_services_vault

### Potential Terraform Configuration

```hcl
resource "azurerm_recovery_services_vault" "rsv" {
  name                = "example-rsv"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"

  notifications = {
      email_subscription_owner = true
      custom_emails = [
          "admin1@example.com",
          "admin2@example.com"
      ]
  }
}
```

**Comment by sean-nixon** [comment]
I'd also be interested in this feature, but support for it might be tricky. Based on configuring this in the Portal, the endpoint for backup notifications appears to be the following:
```
PATCH https://management.azure.com/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.RecoveryServices/vaults/xxx/monitoringConfigurations/notificationConfiguration?api-version=2017-07-01-preview
```

However, I'm unable to find any documentation on this endpoint in the API reference docs and no equivalent client in the azure-sdk-for-go.

**Comment by elliot-resdiary** [comment]
Could this be what we need? https://docs.microsoft.com/en-us/rest/api/site-recovery/replicationalertsettings/create

**Comment by sean-nixon** [comment]
That's specific for Azure Site Recovery replication alerts and doesn't cover backup. Configuring that is equivalent to setting "Email notifications" on "Site Recovery events" in the Portal.

![image](https://user-images.githubusercontent.com/16943522/72618234-49d99180-3900-11ea-85ef-ec8cace6771b.png)

Backup notifications are set in the Portal at "Backup Alerts" > "Configure notifications" and use the seemingly undocumented endpoint I described above if you look in the browser network calls.

![image](https://user-images.githubusercontent.com/16943522/72618263-5a8a0780-3900-11ea-801b-e91eb2b10bac.png)

Your initial issue sounded like you were wanting backup notifications specifically. Regardless, I would love it if Terraform supported both types. Building off your example, a potential configuration could be:

```hcl
resource "azurerm_recovery_services_vault" "rsv" {
  name                = "example-rsv"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"

  replication_notifications = {
      email_subscription_owner = true
      custom_emails = [
          "admin1@example.com",
          "admin2@example.com"
      ]
      locale = "en-US"
  }
 
  backup_notifications = {
      recipient_emails = [
          "admin2@example.com"
       ]
     severities = [
         "Critical",
         "Warning",
         "Information"
     ]
     frequency = "Hourly" # or "Immediate"
  }
}
```

**Comment by hj2m** [comment]
This problem has solutions, who has the solution, please tell me.

**Comment by nitmatgeo** [comment]
Would like to have this enhancement planned :)

**Comment by SteveKurutz** [comment]
I agree that adding support for backup alerting settings is necessary.  Due to the relationships between vaults, regions, and subscriptions, it's necessary to deploy multiple vaults to back up "one" organization.  This results in a whole lot of manual configuration on a vault-by-vault basis.

Similarly, enabling the setting of diagnostic logging in TF would be helpful.  As it stands, we can only automate 80% of what's important to us about vault configuration due to these limitations.
